scheduler/src/modules/transactions/transaction.service.ts

```diff
@@ -0,0 +1,200 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Transaction } from './transaction.entity';
+import { TransactionDetail } from './transaction-detail.entity';
+import { Cron } from '@nestjs/schedule';
+import { LogAsyncMethodExecution } from 'src/common/decorator/log-async-method-execution.decorator';
+import { generate } from 'src/helper/generator.helper';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class TransactionService {
+  constructor(
+    @InjectRepository(Transaction)
+    private readonly transactionRepository: Repository<Transaction>,
+    @InjectRepository(TransactionDetail)
+    private readonly cfsTransactionDetailRepo: Repository<TransactionDetail>
+  ) {}
+
+  async getAllTransactions(): Promise<Transaction[]> {
+    return this.transactionRepository.find();
+  }
+
+  async getTransactionDetails(): Promise<Transaction[]> {
+    const transactions = await this.transactionRepository.createQueryBuilder('transaction')
+    .innerJoinAndSelect('TransactionDetail', 'detail', 'transaction.TRANSACTION_ID = detail.TRANSACTION_ID')
+    .where('detail.CFS_FILE_STATUS_TYPE = :status', { status: 'READY' })
+    .getMany();
+
+  if (transactions.length > 0) {
+    console.log(transactions);
+    return transactions;
+  } else {
+    console.log('No transactions found with status READY');
+    return [];
+  }
+  }
+
+  async updateCfsFileStatusType(fileName: string): Promise<void> {
+    const currentDate = new Date();
+    const currentUTCTimestamp = currentDate.toISOString();
+    const transactionDetails: TransactionDetail[] = await this.cfsTransactionDetailRepo.find({ where: { CFS_FILE_STATUS_TYPE: 'READY' } });
+    const updatedTransactionDetails: TransactionDetail[] = transactionDetails.map(detail => {
+      detail.CFS_FILE_STATUS_TYPE = 'SENT';
+      detail.PROCESSSING_DATE_TIME = currentUTCTimestamp;
+      detail.FILE_NAME = fileName;
+      return detail;
+    });
+
+    try {
+      await this.cfsTransactionDetailRepo.save(updatedTransactionDetails);
+      console.log(`Updated transaction details`);
+    } catch (error) {
+      console.error('Error updating transaction details:', error);
+      throw error;
+    }
+  }
+
+  private currentDate = new Date();
+  private holidays: string[] = [
+    this.getNewYearsDay(), // New Year's Day
```

Comment:
@john-fletcher-aot is looking at if we can use an existing service for the functionality. If it doesn't work out, a holiday calendar table will have to be created in onRouteBC